src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Node/JsonNodeConverter.cs

```diff
@@ -28,6 +28,7 @@ public override void Write(Utf8JsonWriter writer, JsonNode? value, JsonSerialize
             if (value == null)
             {
                 writer.WriteNullValue();
+                // Note the JsonSerializer.Deserialize<T>(JsonNode?) also calls WriteNullValue() for a null + root JsonNode.
```

Comment:
Do you have the right serializer method referenced in this comment?

Reply:
Yes -- the Deserialize() for `JsonNode` has a special [check ](https://github.com/dotnet/runtime/pull/56112/files#diff-b3b917c80d925a0b954eeced485e7ddf6959b907db8069c1b9e4f06380c4bddaR144-R147)for `null` to allow it` to be properly deserialized (since a node is actually nullable, unlike document\element).